FlowCryptAppTests/Functionallity/WKDURLs/PubLookupTests.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  PubLookupTests.swift
+//  FlowCryptAppTests
+//
+//  Created by  Ivan Ushakov on 28.09.2021.
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import XCTest
+import Promises
+@testable import FlowCrypt
+
+final class PubLookupTests: XCTestCase {
```

Comment:
This test is quite trivial (it's hard to imagine any kind of code change that would pass PR review, compile yet break this test), and therefore not very valuable. Of course whatever `KeyDetails` you put into a lookup service is going to come right back, nothing much to test there.

What I'd need testing is parsing a public key with `core.parseKeys` and make sure it returns these two new values. You can try one public key that does not expire (returns nil `expiration`), and one that expires.

This one should expire at `2021-08-31 15:42` and has last valid signature `2020-09-05 15:42`

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: BCPG v1.68

mQINBF9TsfMBEAC+tZIFe6YH7KWTeKCAz+Ii6SzC895jH75K3k3YRlBC3m4Lhena
NY80k82baMmCH13N5E0U5mhp2Qcucsts2n1GLo9j5OXTMg1p7fIKWG3eJh2vYYFi
2CSdwck4e4dvmiAC269L7w6ONVM6opFauyQKOV9O81C+9NBO3eH5uiXTaa1vbVzJ
/KseyZtVGeqkXGEJnEO3s8TgiHXtE9d4dYtDlbuy2ys4x836pwnqs8ChyZpKJfFE
tMV1uinD9yNS7y7ipeuNRJfR7P6WZWR6ZhOe3jKLAb4iAVQwHaFqoLwdNlXuCaUw
JSqdjvIWnSucrohvaLdCdca06Zvx8TieGZ+VlfVWpibOJ0Cf8EbBiU1DCiUv96cQ
6z8d9bCBGWYKdivOFwVWnu2sMCwHmNQAjOV+7091Kn0J77rJr38zGK4RcZ0Z8ZOi
BEMEwYK1ws+gaNlBQJWq+sjwyisJICq0+Iv27+MUnPAxxrCaHmEsz7e1+EGjTZYe
ejxDcaBG6fc4Q5q7S78EjihEV8GfM+bOkRiDG0MCsxYBGBHpdT8c/Ja+VLoMRKue
qDL4MSvWT+runh6Dpik+mq5u+bu95J53d5eUwwht2HY9nnj4X5L3KflkFurO0g2H
XA288bTs6UPCO4f8DrSoeWT7KlVlWlpdrHNLxG2nuqwMdV6BOeJy2EW4kQARAQAB
tCdBbGVqYW5kcm8gQWx2YXJhZG8gPGFsZXhAZmxvd2NyeXB0LmNvbT6JAlMEEAEI
ACYFAl9TsfMFCQHanAAGCwkHCAMCBBUICgIEFgIBAAIZAQIbAwIeAQAhCRC9Bjbx
WJrXmBYhBFJts2bw5J7BHdOJ3L0GNvFYmteY3dsP/jHEqRdeV7npZd1NsAKQb6M+
tkLJYtp6iOh+WiLnLK1/h1ORfPXMLxWoXbah5f3LSxpZviC0fg5Y7AebFhTDgNTo
b5Wd9phky8SctXHzi3rszOYmVscBb+/fCUW36M6n03LbN3rS+opjmndc7EJmysOM
MC2cIKxzufblwDsxh4KaL/OxNn3Wl98Vcl3dTPpytMnokwxw+CwZ8kbpG8iA9JRt
TOO0JI+a+jcjO0QthT+oNM3U8rYBPfQqH4r9SwCuM1b7+Av0Bfj/KMy+qTXLggED
d/oZBxtstoCC8hO9gSJCd5uPR/z6aHb1iBnWze+laAVtThUpiFowkP+rLW7VIMSI
yBsQ657vHKUJEouFMZgZNhECxDrmBx6G/fnUimWl+0yAMYewXaHXtLnsWD8th4vP
wDK9/PRK2xh1Tl8t0K6yxZU70FAA9Ajfli5ijOzqyyKeztzu/JEmg9yxEPxH7HvS
H77BVtN5t/bDWRA6w1C1D5FNRyaghKjJM0M5uTEvGcTTejSKZGsmcXwslibayxTZ
XU6LCVWtI99OpISXjNqX9zRSKLR07i/DgYXSBjYyvkXyDfDOhsC7cK+noAZ/QNeJ
+96dCj/xUf45vad5ZyjqNyqWocnmBY8xMuF/elIZvoFMvTlDAcmUI4Bcdt/f7VRa
HuKprSf6U4tureaV+PLyuQINBF9TsfMBEACfvSlJFkMdjUtsq7t6vKf1vzABhPVO
oyp+kLEjh6MgMdFwxJ3a2P6enezjk+qKm8xMBweO4860zUTC6j6BPeO3rdTim8DM
w8DMfGKDwPbpPTJFON6HLUW3BgIhqYACmmh8KoV1QxLIx1uINrgyH8FrwmNtzznT
a4uqMN+ntYtQfHVX91ICgUU14082wXYwFErCfU/2CfCnc/nqY1/vz1pqrFZ0YXcY
zsfG7094Cf0iF5vWdSWxrvYrs8HRpxWtqQME5XimWGrP68YpOU8cCjbrfLlJx7rv
dHRbF7ESt6vSEOqlojzvOKK5AmSi+BubiYBQZXC8Cb3w4ugHjfeY0VvC+YQakcpd
blLdQJeIOTmJV5UflsIfyZHZfjm0FcxpmrhlVD2Vgk6dEGuWJNvupvjxVzQWZAZ5
33cfSEFFN//ZX4g1RpmeSJYwP5+YKVm0I5RVA82CALGgBW87sLqeP15n0HuDyhn+
7jgleOzHTyLRTjY8t80G/GTMD/WfgfmcQun2GzblytiXm9ALwoVTlqG/B7tB95eV
xfjIXK6kIV+goEaK4NAG2Y19u/Yo+KEQX8IdsHqVHyY/cYKqq7a4s8WsKaaFG39e
QhOq26Bd/ky5FwjWuk3T3LndXee6NuwgrlkNBxDhzUebJphso3Fnjb9NAtCKujIG
MEAi1iJLSx3CaQARAQABiQI8BBgBCAAPBQJfU7HzBQkB2pwAAhsMACEJEL0GNvFY
mteYFiEEUm2zZvDknsEd04ncvQY28Via15iIJg/+Lc3w9KAwq74nM+EOPZaln4O0
SUWSoWPv3E43cUYyvlnkeB/tNHHJ/+4VyDyWAVQGkF/lN8WiWMhv39oz7SXKlXjR
srh9S6+oVouo1reYsPwO6g9HeumvNdL00zPKWgUe9NsKGepiejPSeUWYQs1QWZBL
yCa+o54/XxX5D/hWvjJWUv1bzhEMdakp3CgAE9ClFHd+KwCuZfoeLyw3f4sc3yEK
ZhVZs0x5yti39HoEF1faWWI1xRz3pYZ582UW6HHwEoFoRxQwm/1+F+CAZPJRyU6j
49qsvetDdKsfumQ5S8BfDnyXDH/eBecwtBIoztFKmg56xN1OzQfwGIcVO/VC8Nir
P8BClfjsUf2vaD+XbzlmSvt69BC7Z6x6nxf2gDZiaKkUFESQ+0cyFn4eyG39rOn5
yEqmtLdoIymOK2B5Wajvc0R0zuwC9jCSZbtBYwHgWTCpXOSVeuFN66nPgRsZAti+
yg5zbnVVzhartgdE/jGgYdza5qMzYDsA5WmKDynpv4dxaX95cuHGDUYUHFPKiOjT
X2cApJ/D4+5dOaJgsGzHCWVw+HEF5RrDhB8YeVs2wPSAsdH5uK7gwKT4tCTAYcTt
n5dng5/K2eE2Ltq39T3zG6QzoxK/AQnTfjLAnj2eNQzqKNmIzzDD4mAprchorO6A
HCUbBA3weLi2IQDZeAA=
=xdFY
-----END PGP PUBLIC KEY BLOCK-----
```

This one below does not expire. Last sig `2017-11-02 17:59`:

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: BCPG v1.68

mQINBFn7XPcBEAC4vqCA+3+voI35TvvrUZqMclLV9NFcgx0fyT+YXpMjbUZRd0+L
FB+4took74VLlHIkyTGO8mqleXdT+2/8v+TWAeIYYii7eVWRWwxn2sfMSqeEEKsg
Qv2whosQRrWdE7DrT6O1VxOvBxN5ZVx7iBBglruqCCpHP3zPgQw1mwdFtJHIS3tF
wQ66ojvcbWznX/yr3qRG1zUzTrvf1kSWr5ZRBRb7Ti8FF9Wa70RraxF2wBa2LNwO
1AwUvpTl1el3Fm9tUQnPAcw53R+HdZzxYjvyhHoJYJu107NKgTmwkDyLcxm7MRJK
h9SOSjrFAA2cgp0W4SSCYUpVyCqUclYRv09XXg0S2zXHJLlLJ8cSlM6jsZnq7jUU
Vqgbts3gmusGHjkuMMYbj3L6U+/gAHQ0kjAcz1Zfh0CWKz2C0aLyl4pxRQh9XUwC
AUdR8Hcz9i4TfAFpVCAicCXR+2wWcy1zzKmtRuqmWi2WdgJpC3hr774QeXZQ4m7j
MiR6fwCRz0tb7OYMVdFB9UShWcpvAq6DPsjTfMz6NWJ1izi8umy2wDhvncTa4nVi
12TQORSNVnsMGZiJkl0GZbsnYXugPrnCirtZlGuCoIwI4iVFIbn6B7hZ+7+FXN/7
Jtc1pvGb0GBtJWZR95wLfFSAFwRSc0zkPfYGjQCDjrKl5n/tJIKqv0botQARAQAB
tChIdW1hbiBhdCBGbG93Q3J5cHQgPGh1bWFuQGZsb3djcnlwdC5jb20+iQI/BBAB
CAApBQJZ+1z5BgsJBwgDAgkQDJwuak0nPG8EFQgKAgMWAgECGQECGwMCHgEACgkQ
DJwuak0nPG/v4w/+Lkl5jbCjpd7IiJgRjnPyo8XGdIs/Ero+BEcqkmRSvvRLURLl
oYM4IbUq38DTOeMpvrerr3mugTQy5N5KLsis9tKRSMhh6wVci3Zz9qnETG6NW0sk
vCi16I8yOjYFuSZyhG5s/GvKWfCe2BE4tKyp9ywCzkl4L5RrG5XpgvlnDXU9aKQh
gMO6cIf5d0R38sq5jnC+bn0dhljyE0Qq29SG/jAYm7O5sLRzZHB3dOhDxh3ZEyuv
0oO9/NQt3g4HBJ0XrUG3BiSah4YP/bDMwsJQ/NcfVYUmZtNVP9vrQxC09gz+hT+Q
WPtd6yGLqHciwzjI/KJ7vbziOB5EHYZ0O36p5E4ekTAg+HEnVjKxGc1pCdaxIWp9
KZ1lbLnhr7RMFVCoV+1QzlvrXQcI3DZtpzTEKhbcALaWoKFB0nB7RpxDIEzPmncw
YAzaozD7JSkZqQErAS7h/L07Kjzlv0tTjPUCcY6D9VQwAuD+ikJm45obedQOU0zT
DiItFyHtcbemoIWiBr5YKShWT8vufm3JkA6e/1saOGDnlylwWtoVDC5dEeFnpZS9
87AfQkRM9Q6/TNtFXwVHmAbbTYe6zSvWYUlBLVc20yJZvVeNEMOHj+npl7L19Kmy
aRME0SYOg387w1WGdDTtXFLj7GU+NtGBjUz0dLSl9dWscYso9bhK2FmS28C5Ag0E
Wftc9wEQAMsHieiVC0mtk6KjURw2ywlX+2roaNhHm2XGYnyhM+Um8XeSPHFZ3u79
65eIrN9lXxDbTtrMHn6tyuVPlPASrnVD0w5/epIe7fKI8U+8Ghpkz4vFqcbry2Ty
DLKGrZpDb9SOSkbeSAFgxgJUAPBWjC9Md4eEYIccdPUMR0E9KII61HJAKZG1O2nx
aXVmV8RTnpt1DqHWUIqj7EFUgHvJAo2LZTZN7PmQAXIBWHkiF14vUrjMvg8aOC9Y
uGbuWBoW8Ywui4DdV8uNSNJmm4YhdV0panvrzRvH4sYuUeLx96BcFSA4hBz1TPGB
sgLJP6QbnCJXx3Ksp0sPvEah88Ot7z+80d9D4EKcRuZF1Oru8pCxTRLOyQ/Ii/98
nEOUmfi6bQpPtUzzAxlDXxu6orSdfTYU/TNvXwOKolTv9UvV/U+eWGO3X0JQmPRa
7R19y9qSlJlKt8mIpLBrWf6fS2WaVcJeLEVu1ujKeeP2i+MkeUeDCTWc9pyV0dUu
fYmVjvk62IH3mTw6GPJeqytkBD4Lt63QV0DRh0g7mWaEiNxwGWZ4dkDfhAsoNKud
RSCXQWcGKrgUljU4J4qlnMh57pnKjDBdwsr/vkFawopG/jlCGKKzF4dWqHUCNj9N
Yyeu8bHmCkWZWl0W4ZfA0WkBbUEerOyDYs8o1YYvXnM9AG3747CZABEBAAGJAikE
GAEIABMFAln7XPoJEAycLmpNJzxvAhsMAAoJEAycLmpNJzxv198P+wVYxGEAugzK
OPSEHUrdB4/qeSUJ2ZdO0JsYi5D1YdhQB5XGK9kWplcv80qiggM5xVt3/NY+09GF
RYYH2s6B+qk6uJA4vD0AD4DLM0Zucys02leqwrj2BCpccvQ71QDxpkBrWAWRhjzS
MxI2gD8rDSa2W/63/kGsG3fo7/4TWzOFFyIUQLrgFj+yiWzoLwiiM+qiyjU8Qjxt
A02Y7P0/Tyo9IMc5ZujReGaqj9hi15Xum/eOeZjxE3N4rhBtnSK0R0KvIKG3DBhO
LaZv4bcp5LKX6qZj4uHTFp2/cz2IaXHdAFwQnWAHVkIGphu/IoGbetdm21eNKdiR
Ypkjgunv4puGS4LcSv84S/fTGOX5gRIx14Pne8RExTPWZzAK3C1VTkp+Z8xZ/X2m
2OeAgy51DZbyjvyosxnH1RhHDKcgcA5gruU+MwjHrv9rHiKD5j3AiXS0B6sTl5dA
z7B7o74IY5Xczqt70R2yWvZLquaCS5YLDDNlGGYBK6tuK8ls8lg3KLFPX1AZnUo8
RImGvQlU63bqKnuotwPlGOXPR1W2lVuTA/GqO86TV8F4NdouvSGHvtGJOyIt+hXx
Y6/iYqbWOyrR0iHIOp/2d7MlvZNyGGkMT12rt1V9fZUcQo3BHJnA7p79ZCLMgch9
wCLordqI7sx/M2kHNWkXd2/SiWG076sC
=BCaS
-----END PGP PUBLIC KEY BLOCK-----
```

Reply:
After a second look, now I see you're trying to test code that's inside `private func parseKey`, which is not very convenient to test as is - I understand. Still, the above is a weak test.

To test this more conveniently, I'd recommend to also address this issue in the same PR: https://github.com/FlowCrypt/flowcrypt-ios/issues/557 - if you make that a public extension, then you can test it conveniently independently, along with what I recommend above. That will be a much better test.

Reply:
Sorry for the harsh review above - it's not as bad as I originally thought, as it does test some of the code you updated - but it doesn't quite test the larger picture (all the way from TypeScript to Swift), which is what I'd like to test.